### PR TITLE
SQL-165 Add CORS for CF template

### PR DESCRIPTION
**Pull request by milt**
[SQL-165] The CF template needs some love, specifically in the form of CORS settings.

The logic is now as follows: If the template is given the CORSAllowedOrigins param, these are used. If not, the ALBHostname is used, if it is not present no web origins are allowed.

[SQL-165]: https://yet.atlassian.net/browse/SQL-165?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: dev-resources/template/2_lrs.yml [hunk 1]
```diff
@@ -376,25 +382,36 @@ Resources:
...
-                }
+                - AllowedOrigins: !If
+                    - SetCORS
+                    - !Sub
+                      - '["${JoinedAllowedOrigins}"]'
+                      - JoinedAllowedOrigins: !Join ['","', !Ref CORSAllowedOrigins]
+                    - !If
+                      - SetDNS
+                      - !Sub '["https://${ALBHostName}"]'
+                      - "[]"
```
Comment (kelvinqian00) [inline dev-resources/template/2_lrs.yml:414]: What exactly does this section mean in plain English?

Comment (milt) [reply dev-resources/template/2_lrs.yml:414]: See the description above:

Comment (milt) [reply dev-resources/template/2_lrs.yml:414]: > The logic is now as follows: If the template is given the CORSAllowedOrigins param, these are used. If not, the ALBHostname is used, if it is not present no web origins are allowed.

Comment (milt) [reply dev-resources/template/2_lrs.yml:414]: This info is also provided in the parameter description.

File: dev-resources/template/2_lrs.yml [hunk 2]
```diff
@@ -93,7 +93,7 @@ Parameters:
   LrsVersion:
     Description: Version of SQL LRS to download and install (public release versions on GitHub)
     Type: String
-    Default: v0.3.9
+    Default: v0.5.1
```
Comment (cliffcaseyyet) [inline dev-resources/template/2_lrs.yml:96]: yay

**Review by cliffcaseyyet — APPROVED** [review]

**Review by kelvinqian00 — APPROVED** [review]
